src/gui/dialogs/campaign_selection.cpp

```diff
@@ -339,6 +340,27 @@ void campaign_selection::pre_show(window& window)
 		page_ids_.push_back(campaign["id"]);
 	}
 
+	std::vector<std::string> dirs;
+	filesystem::get_files_in_dir(game_config::path + "/data/campaigns", nullptr, &dirs);
+	if(dirs.size() <= 1) {
```

Comment:
Maybe 5 or 10 here, so the message still shows up if someone's told the software center to install Heir to the Throne, and `wesnoth-httt` has pulled in `wesnoth-core`. That would also cover some service dropping `Thumbs.db` files all over the filesystem.

Reply:
If someone manually installed an additional campaign, then they would know that there are more to install already.

This also only gets a list of the directories, so a Thumbs.db file wouldn't get included in that count.

Reply:
Not necessarily. They might believe that there's only the tutorial, the single official campaign, and the add-ons server.

Reply:
How would they only find out about a single campaign package and not any of the others?

Reply:
I don't think you should underestimate the strange things that people might do. As a random example, maybe it's possible that their search in the package manager happened to match a keyword for one of the campaigns, and they installed that campaign without realizing it was just one of several for the base game. Obviously installing the campaign would also pull in the base game as a dependency, and just like in Stevecotton's example, they'd have a game with only one of the official campaigns plus the tutorial.

Why not just make the internal number match the number shown to the users? That is, if there are fewer than 16 campaigns (15 plus the tutorial), show the message.

Reply:
Alright, updated then.

Reply:
Regarding the 'updated' reply - maybe I'm mistaken but it looks like the final commit still says `<= 1` directories.

Reply:
It's an additional commit pushed separately to master after merging: d492a36019eaddd9ed3a664b25f31887615f0dc7